From 52a2d433548c06ae1d27fed5017fc49ce4a61eea Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 9 Sep 2018 13:37:21 +0200
Subject: [PATCH 1/1] efi_loader: console input ESC a - ESC z

We have foreseen a work around for entering control codes in the
EFI_SIMPLE_TEXT_INPUT_PROTOCOL. But currently we have an offset of one.

ESC a should translate to 0x01 (CTRL+a).
ESC z should translate to 0x1a (CTRL+z).

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_console.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_console.c b/lib/efi_loader/efi_console.c
index 3ca6fe536c..5701782663 100644
--- a/lib/efi_loader/efi_console.c
+++ b/lib/efi_loader/efi_console.c
@@ -484,7 +484,12 @@ static efi_status_t EFIAPI efi_cin_read_key_stroke(
 			pressed_key.scan_code = ch - 'P' + 11;
 			break;
 		case 'a'...'z':
-			ch = ch - 'a';
+			/*
+			 * Workaround for entering CTRL+a (0x01) - CTRL+z (0x1a)
+			 * as escape sequence if the terminal does not allow
+			 * direct entry. These are not Xterm control sequences.
+			 */
+			ch = ch - 'a' + 1;
 			break;
 		case '[':
 			ch = getc();
-- 
2.18.0

